sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs

```diff
@@ -50,7 +50,7 @@ public virtual Response<Key> CreateKey(string name, KeyType keyType, KeyCreateOp
         public virtual async Task<Response<Key>> CreateKeyAsync(string name, KeyType keyType, KeyCreateOptions keyOptions = default, CancellationToken cancellationToken = default)
         {
             if (string.IsNullOrEmpty(name)) throw new ArgumentException($"{nameof(name)} can't be empty or null");
-            if (keyType == default) throw new ArgumentNullException(nameof(keyType));
+            if (string.IsNullOrEmpty(keyType.StringValue)) throw new ArgumentNullException(nameof(keyType));
```

Comment:
>throw new ArgumentNullException(nameof(keyType)); [](start = 59, length = 49)

It's not really a null argument in this case.  Perhaps it should just be an ArgumentException with a more descriptive message.